src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -1,22 +1,64 @@
 package frc.robot.subsystems;
 
-import com.ctre.phoenix.motorcontrol.FollowerType;
+import com.ctre.phoenix.motorcontrol.ControlMode;
 import com.ctre.phoenix.motorcontrol.InvertType;
 import com.ctre.phoenix.motorcontrol.VictorSPXControlMode;
-import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+
 import static frc.robot.Constants.*;
 
-public class IntakeSubsystem {
-    private VictorSPX upperIntakeMotor = new VictorSPX(UPPER_INTAKE_MOTOR);
-    private VictorSPX bottomIntakeMotor = new VictorSPX(BOTTOM_INTAKE_MOTOR);
+public class IntakeSubsystem extends SubsystemBase {
+    private WPI_VictorSPX upperIntakeMotor = new WPI_VictorSPX(UPPER_INTAKE_MOTOR);
+    private CANSparkMax bottomIntakeMotor = new CANSparkMax(BOTTOM_INTAKE_MOTOR, MotorType.kBrushed);
+    private DigitalInput intakeSwitch = new DigitalInput(INTAKE_SWITCH);
+    private DoubleSolenoid intakeSolenoid = new DoubleSolenoid(INTAKE_SOLENOID_FWD, INTAKE_SOLENOID_REV);
+    private Trigger intakeSwitchButton = new Trigger(intakeSwitch::get);
+
+    private int ballsCollected = 0;
+    private boolean isAlreadyPushed = false;
 
     public IntakeSubsystem() {
-        upperIntakeMotor.follow(bottomIntakeMotor);
-        upperIntakeMotor.setInverted(InvertType.OpposeMaster);
+        // upperIntakeMotor.setInverted(InvertType.OpposeMaster);
+        intakeSwitchButton.whenActive(new InstantCommand(() -> { 
+            ballsCollected++;
+            System.out.println("Ball count: " + ballsCollected);
+        }));
+    }
+
+    // Solenoid methods
+
+    public void extend() {
+        intakeSolenoid.set(DoubleSolenoid.Value.kReverse);
     }
 
-    public void succySuccy(double succySpeed) {
-        bottomIntakeMotor.set(VictorSPXControlMode.PercentOutput, succySpeed);
+    public void spin(double speed) {
+        upperIntakeMotor.set(VictorSPXControlMode.PercentOutput, speed);
```

Comment:
In python we would have a variable that holds the speed and then in the periodic method it would feed that speed to the motor. Is this the recommended way for command based?

Reply:
See that's what I'm not sure of... I've looked at the weird frameworks that teams like 254 and others use, and while it doesn't seem like they do exactly what magicbot would do (output from variables then reset to 0 every loop), they have some kind of periodic signal output code.